### ISSUE TITLE
pack_rows() with tabular = "tabularx" returns error

### ISSUE THREAD

**Issue by thmschk** [haozhu233/kableExtra#861]
I am unable to use pack_rows() in combination with `tabular = "tabularx"`, which produces an error. 

```
library(kableExtra)
library(dplyr)
kbl(mtcars, format="latex", tabular = "tabularx", valign="{\\textwidth}") %>% kableExtra::pack_rows("XXX", 1,2)

# `error in seq.default(table_info$ncol) : 'from' must be a finite number`
```

**Comment by thmschk** [comment]
I dived into the code a bit to understand the problem and guess that `magic_mirro_latex()` can't handle tabularx.  As far as I can see, the ifelse condition might be extended to include tabularx:

```{r}
table_info$tabular <- ifelse(grepl("\\\\begin\\{tabular\\}", kable_input), "tabular", ifelse(grepl("\\\\begin\\{tabularx\\}", kable_input), "tabularx", "longtable"))

```

**Comment by dmurdoch** [comment]
Fixed by #870.

**Comment by dmurdoch** [comment]
Thanks for the report and fix!
